Tasks/AzureRmWebAppDeployment/operations/KuduServiceUtility.ts

```diff
@@ -0,0 +1,323 @@
+            }
+        }
+        tl.debug('postDeployment script path to execute : ' + scriptPath);
+        return {
+            filePath: scriptPath,
+            isCreated: false
+        }
+    }
+
+    private async _warFileDeployment(packagePath: string, physicalPath: string, virtualPath?: string): Promise<string> {
+        tl.debug('WAR: webAppPackage = ' + packagePath);
+        let warFile = path.basename(packagePath.slice(0, packagePath.length - '.war'.length));
+        let warExt = packagePath.slice(packagePath.length - '.war'.length)
+        tl.debug('WAR: warFile = ' + warFile);
+        warFile = warFile + ((virtualPath) ? "/" + virtualPath : "");
+        tl.debug('WAR: warFile = ' + warFile);
+        physicalPath = physicalPath + "/webapps/" + warFile;
+        await this.createPathIfRequired(physicalPath);
+        return physicalPath;
+
+    }
+
+    private async _appOfflineKuduService(physicalPath: string, enableFeature: boolean): Promise<void> {
+        if(enableFeature) {
+            tl.debug('Trying to enable app offline mode.');
+            var appOfflineFilePath = path.join(tl.getVariable('AGENT.TEMPDIRECTORY'), 'app_offline_temp.htm');
+            tl.writeFile(appOfflineFilePath, '<h1>App Service is offline.</h1>');
+            await this._appServiceKuduService.uploadFile(physicalPath, 'app_offline.htm', appOfflineFilePath);
+            tl.debug('App Offline mode enabled.');
+        }
+        else {
+            tl.debug('Trying to disable app offline mode.');
+            await this._appServiceKuduService.deleteFile(physicalPath, 'app_offline.htm');
+            tl.debug('App Offline mode disabled.');
+        }
+    }
+
+    private async _pollForFile(physicalPath: string, fileName: string, noOfRetry: number): Promise<void> {
+        var attempts: number = 0;
+        if(tl.getVariable('appservicedeploy.retrytimeout')) {
+            noOfRetry = Number(tl.getVariable('appservicedeploy.retrytimeout')) * 6;
+            tl.debug('Retry timeout provided by user: ' + noOfRetry);
+        }
+        tl.debug('Polling started for file: ' + fileName);
+
+        while (attempts < noOfRetry) {
+            attempts += 1;
+            var fileContent: string = await this._appServiceKuduService.getFileContent(physicalPath, fileName);
+            if(fileContent == null) {
+                tl.debug('File: ' + fileName + ' not found. retry after 10 seconds. Attempt: ' + attempts);
+                await webClient.sleepFor(10);
+            }
+            else {
+                tl.debug('Found file:  ' + fileName);
+                return ;
+            }
+        }
+
+        if(attempts == noOfRetry) {
+            throw new Error(tl.loc('ScriptStatusTimeout'));
+        }
+    }
+
+    private _getUpdateHistoryRequest(isDeploymentSuccess: boolean, deploymentID?: string, customMessage?: any): any {
+        
+        var status = isDeploymentSuccess ? 4 : 3;
```

Comment:
Should we define some constants for successcode as numbers are not understandable directly. Please define like const int SUCCESS_CODE = 4 and use constant here.

Reply:
sure. will do it.

Reply:
done